lib/galaxy/managers/collections.py

```diff
@@ -252,6 +256,27 @@ def __recursively_create_collections( self, trans, element_identifiers ):
 
         return element_identifiers
 
+    def __recursively_create_collections_for_elements( self, trans, elements ):
+        if elements is self.ELEMENTS_UNINITIALIZED:
+            return
+
+        new_elements = odict.odict()
+        for key, element in elements.iteritems():
```

Comment:
Can elements be so large that iteritems() and an iterator is needed. In the spirit of your Python3 effort items() would be better, imho.